.changeset/fresh-keys-enjoy.md

```diff
@@ -0,0 +1,42 @@
+---
+'@xstate/store': minor
+---
+
+Added store getters API for computed values derived from context:
+
+```ts
+const store = createStore({
+  context: { count: 2 },
+  getters: {
+    doubled: (ctx) => ctx.count * 2,
+    squared: (ctx) => ctx.count ** 2,
+    // Can depend on other getters (types can not be inferred, due to circular references)
+    sum: (ctx, getters: { doubled: number; squared: number }) =>
+      getters.doubled + getters.squared
+  },
+  on: {
+    inc: (ctx) => ({ count: ctx.count + 1 })
+  }
+});
+
+// Getters are available on store snapshots
+var snapshot = store.getSnapshot();
+assert.equal(snapshot.doubled, 4);
+assert.equal(snapshot.squared, 4);
+assert.equal(snapshot.sum, 8);
```

Comment:
Not sure yet how I feel about getters being directly on the `snapshot` object, especially when we have to prevent collisions with the existing properties/methods

Reply:
Sure I can move it out.

Some questions
1. What does the snapshot represent?
2. What do you feel about the vocabulary of "getters"?
3. Is there anything I am obviously missing here?

Reply:
The snapshot contains the:
- `context` (contextual data for the snapshot)
- `status` (the status of the actor itself, such as `"active"` or `"stopped"` or `"error"` or `"done"`)
- `output` (the "done data" for when the actor is `"done"`)
- `error` (the error data for when the actor has an `"error"` status)

Reply:
The pattern I was trying to replicate I pulled from Vue universe: https://pinia.vuejs.org/core-concepts/getters.html

Reply:
Nested the getters.